Src/EmbeddedScripts.JS.ChakraCore/TypeMapper.cs

```diff
@@ -1,13 +1,18 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
 using ChakraHost.Hosting;
 
 namespace EmbeddedScripts.JS.ChakraCore
 {
-    public class TypeMapper
+    public class TypeMapper : IDisposable
     {
         private readonly JsContext _context;
+        
+        // stores registered delegates to prevent their garbage collection
+        private readonly List<JavaScriptNativeFunction> _jsNativeFunctions = new();
```

Comment:
nit: you can use ConcurrentBag as you don't need a list with preserved order here